test/test_maskedtensor.py

```diff
@@ -96,6 +105,195 @@ def _fix_fn_name(fn_name):
     return fn_name
 
 
+class TestBasics(TestCase):
+    def test_add(self):
+        data = torch.arange(5.0)
+        mask = torch.tensor([True, True, False, True, False])
+        m0 = MaskedTensor(data, mask)
+        m1 = MaskedTensor(data, ~mask)
+        self.assertRaises(ValueError, lambda: m0 + m1)
+
+    def test_softmax(self):
+        x = torch.randn(3, 4) * 0.1
+        m = torch.tensor(
+            [
+                [True, True, True, False],
+                [False, True, False, True],
+                [True, True, False, False],
+            ]
+        )
+        mx = MaskedTensor(x, m, requires_grad=True)
+        ts = torch.softmax(mx, -1)
+        ts.sum().backward()
+        xinf = x.masked_fill(~m, float("-inf")).detach().clone().requires_grad_()
+        torch.softmax(xinf, -1)
+
+    def test_where(self):
+        # http://pytorch.org/maskedtensor/main/notebooks/nan_grad.html
+        x = torch.tensor([-10.0, -5, 0, 5, 10, 50, 60, 70, 80, 90, 100])
+        mask = x < 0
+        mx = MaskedTensor(x, mask, requires_grad=True)
+        my = MaskedTensor(torch.ones_like(x), ~mask, requires_grad=True)
+        y = torch.where(mask, torch.exp(mx), my)
+        s = y.sum()
+        s.backward()
+
+    def test_mha_issue_41508(self):
```

Comment:
this isn't a sufficient test of the MT `multi_head_attention_forward()`. we need to either fully test it a la the core, non-MT version, or not ship it.

the fact that this is the only test makes me think that its purpose is basically a demo of how MTs can be used to fix 41508. That's cool but it doesn't motivate adding it to core - throwing yet another mha function in there is counterproductive, it just confuses things wrt what mha function to use. Personally I'd say we should remove it but @albanD @cpuhrsch might have other opinions.

Reply:
removed for now as per our offline discussion